host-go/repository/txn_registry.go

```diff
@@ -0,0 +1,104 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package repository
+
+import (
+	"context"
+
+	"github.com/lens-vm/lens/host-go/config/model"
+	"github.com/sourcenetwork/immutable/enumerable"
+)
+
+type implicitTxnRepository struct {
+	repository *repository
+	db         TxnSource
+}
+
+type explicitTxnRepository struct {
+	repository *repository
+	txn        Txn
+}
+
+var _ Repository = (*implicitTxnRepository)(nil)
+var _ Repository = (*explicitTxnRepository)(nil)
+
+func (r *implicitTxnRepository) Init(txnSource TxnSource) {
+	r.db = txnSource
+}
+
+func (r *explicitTxnRepository) Init(txnSource TxnSource) {}
+
+func (r *explicitTxnRepository) WithTxn(txn Txn) Repository {
```

Comment:
suggestion: `WithTxn` on `implicitTxnRepository` was removed at one point. If you want to eventually make the explicit transaction work, this will be needed.

Reply:
I can confirm that it was :)

I'm ok with that. We'll just have to figure out a way to reintroduce explicit transactions at some point if we care to have lens respect them.